python_dwd/file_path_handling/file_list_creation.py

```diff
@@ -1,75 +1,46 @@
                                     parameter: Union[Parameter, str],
                                     time_resolution: Union[TimeResolution, str],
                                     period_type: Union[PeriodType, str],
-                                    folder: str = DWD_FOLDER_MAIN,
-                                    create_new_filelist=False) -> pd.DataFrame:
+                                    create_new_file_index: bool = False) -> pd.DataFrame:
     """
     Function for selecting datafiles (links to archives) for given
     station_ids, parameter, time_resolution and period_type under consideration of a
     created list of files that are
     available online.
-
     Args:
-        station_ids: id(s) for the weather station to ask for data
+        station_id: id for the weather station to ask for data
         parameter: observation measure
         time_resolution: frequency/granularity of measurement interval
         period_type: recent or historical files
-        folder:
-        create_new_filelist: boolean for checking existing file list or not
-
+        create_new_file_index: set if new file index is created
     Returns:
         List of path's to file
-
     """
+    if create_new_file_index:
+        reset_file_index_cache()
+
     parameter = Parameter(parameter)
     time_resolution = TimeResolution(time_resolution)
     period_type = PeriodType(period_type)
 
-    # Check type of function parameters
-    station_ids = [int(statid) for statid in station_ids]
-
-    # Check for the combination of requested parameters
-    check_parameters(parameter=parameter,
-                     time_resolution=time_resolution,
-                     period_type=period_type)
-
-    # Create name of fileslistfile
-    filelist_local = f'{FILELIST_NAME}_{parameter.value}_' \
-                     f'{time_resolution.value}_{period_type.value}'
-
-    # Create filepath to filelist in folder
-    filelist_local_path = Path(folder,
-                               DWD_FOLDER_METADATA,
-                               filelist_local)
-
-    filelist_local_path = f"{filelist_local_path}{DATA_FORMAT}"
-
-    if create_new_filelist or not Path(filelist_local_path).is_file():
-        create_fileindex(parameter=parameter,
-                         time_resolution=time_resolution,
-                         period_type=period_type,
-                         folder=folder)
+    file_index = create_file_index_for_dwd_server(
+        parameter, time_resolution, period_type)
 
-    filelist = pd.read_csv(filepath_or_buffer=filelist_local_path,
-                           sep=",",
-                           dtype={DWDMetaColumns.FILEID.value: int,
-                                  DWDMetaColumns.STATION_ID.value: int,
-                                  DWDMetaColumns.FILENAME.value: str})
+    file_index = file_index[
+        file_index[DWDMetaColumns.STATION_ID.value] == int(station_id)
```

Comment:
if station_id is a list you can check with file_index[DWDMetaColumns.STATION_ID.value] .isin()

Reply:
See above